checkov/secrets/plugins/entropy_keyword_combinator.py

```diff
@@ -269,6 +275,14 @@ def analyze_line(
 
         return set()
 
+    def postprocess_secrets(self, detected_secrets: set[PotentialSecret], line: str) -> None:
+        for detected_secret in detected_secrets:
+            if detected_secret.secret_value and line.replace('"', '').replace("'", '').startswith(
+                    detected_secret.secret_value):
+                # Found keyword prefix as potential secret
+                detected_secrets.remove(detected_secret)
+                break
```

Comment:
@maxamel why not go over the entire secret? Why just on the first one?

Reply:
Because after you cut the first part, if you continue iterating you'll just remove all the potential secrets and end up with nothing. We're interested only in the prefix since that indicates a keyword has been flagged.